src/software/ai/hl/stp/tactic/attacker/attacker_fsm.h

```diff
@@ -7,6 +7,45 @@
 #include "software/ai/hl/stp/tactic/tactic.h"
 #include "software/ai/intent/move_intent.h"
 #include "software/ai/passing/pass.h"
+#include "software/ai/evaluation/find_open_areas.h"
+
+MAKE_ENUM(AttackerFSMStates,
+        KEEP_AWAY,
+        SHOOTING,
+        PASSING
+        )
+
+struct PasserFSM
+{
+    using ControlParams = PivotKickFSM::ControlParams;
+    // this struct defines the only event that the PivotKickFSM responds to
+    DEFINE_UPDATE_STRUCT_WITH_CONTROL_AND_COMMON_PARAMS
+
+    auto operator()() {
+        using namespace boost::sml;
+
+        const auto pass_s = state<PivotKickFSM>;
+        return make_transition_table(
+                *pass_s = X
+                        );
+    }
+};
+
+struct ShooterFSM
```

Comment:
Because we split the `PivotKickFSM` state, we now had 2 states that needed the same sub-FSM, which boost::sml doesn't allow.

```
const auto shoot_s = state<PivotKickFSM>;
const auto pass_s = state<PivotKickFSM>;
```
doesn't work because they're technically the same type. I got stuck for quite a while on this bug, because it fails very subtly and messes things up.

So we had to basically wrap the `PivotKickFSM` in new types so we could use it in 2 different states. This could probably be made a macro if you need it more often.